purchase_work_acceptance/models/purchase.py

```diff
@@ -0,0 +1,96 @@
+        comodel_name='work.acceptance.line',
+        inverse_name='purchase_line_id',
+        string="WA Lines",
+        readonly=True,
+    )
+
+    def _compute_wa_ids(self):
+        for order in self:
+            order.wa_ids = \
+                order.mapped('order_line').mapped('wa_line_ids').mapped('wa_id')
+            order.wa_count = len(order.wa_ids)
+
+    @api.multi
+    def action_view_wa(self):
+        self.ensure_one()
+        act = self.env.ref('purchase_work_acceptance.action_work_acceptance')
+        result = act.read()[0]
+        create_wa = self.env.context.get('create_wa', False)
+        result['context'] = {
+            'default_purchase_id': self.id,
+            'default_partner_id': self.partner_id.id,
+            'default_company_id': self.company_id.id,
+            'default_currency_id': self.currency_id.id,
+            'default_date_due': self.date_planned,
+            'default_wa_line_ids': [(0, 0, {
+                'purchase_line_id': line.id,
+                'name': line.name,
+                'product_uom': line.product_uom.id,
+                'product_id': line.product_id.id,
+                'price_unit': line.price_unit,
+                'product_qty': line._get_product_qty(),
+                }) for line in self.order_line],
+        }
+        if len(self.wa_ids) > 1 and not create_wa:
+            result['domain'] = "[('id', 'in', " + str(self.wa_ids.ids) + ")]"
+        else:
+            res = self.env.ref(
+                'purchase_work_acceptance.view_work_acceptance_form', False)
+            result['views'] = [(res and res.id or False, 'form')]
+            if not create_wa:
+                result['res_id'] = self.wa_ids.id or False
+        return result
+
+    @api.multi
+    def action_view_invoice(self):
+        if self.env.context.get('create_bill', False) and self.env.user.has_group(
+                'purchase_work_acceptance.group_work_acceptance'):
+            wizard = self.env.ref(
+                'purchase_work_acceptance.view_select_work_acceptance_wizard')
+            return {
+                'name': _('Select Work Acceptance'),
+                'type': 'ir.actions.act_window',
+                'view_type': 'form',
+                'view_mode': 'form',
+                'res_model': 'select.work.acceptance.wizard',
+                'views': [(wizard.id, 'form')],
+                'view_id': wizard.id,
+                'target': 'new',
+            }
+        return super(PurchaseOrder, self).action_view_invoice()
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    wa_line_ids = fields.One2many(
+        comodel_name='work.acceptance.line',
+        inverse_name='purchase_line_id',
+        string="WA Lines",
+        readonly=True,
+    )
+
+    def _get_product_qty(self):
+        if self.product_id.type == 'service':
+            return self.product_qty - self.qty_invoiced
+        return self.product_qty - self.qty_received
```

Comment:
Why? I don't understand this.

Reply:
Because product type = service don't have  qty_received.
But I will add qty_received follow https://github.com/OCA/purchase-workflow/pull/774#discussion_r345090165, this code will change.

Reply:
Edited : https://github.com/OCA/purchase-workflow/pull/774/commits/74b8c5f27338d714143dfee5372105bf91e269cc